cognite/neat/data_model/models/entities/_base.py

```diff
@@ -0,0 +1,100 @@
+from functools import total_ordering
+from typing import Annotated, Any
+
+from pydantic import BaseModel, Field, field_validator, model_serializer
+
+from ._constants import (
+    Undefined,
+    Unknown,
+    _UndefinedType,
+    _UnknownType,
+)
+
+
+@total_ordering
+class Entity(BaseModel, extra="ignore", populate_by_name=True):
+    """Entity is a concept, class, property, datatype in semantics sense."""
+
+    prefix: (
+        Annotated[str, Field(pattern=r"^[a-zA-Z][a-zA-Z0-9_-]{0,41}[a-zA-Z0-9]?$", min_length=1, max_length=43)]
+        | _UndefinedType
+    ) = Undefined
```

Comment:
Wouldn't this be simpler?
```suggestion
    prefix: str |  _UndefinedType = Field(default=_UndefinedType, pattern=r"^[a-zA-Z][a-zA-Z0-9_-]{0,41}[a-zA-Z0-9]?$", min_length=1, max_length=43)
```